### PR TITLE
Feature/etch 375 radio

### PR DESCRIPTION
**Pull request by gavmck**
Screenshot 2024-02-16 at 06 08 36
Screenshot 2024-02-16 at 06 08 41

A radio checkbox component because a radio and a checkbox is virtually the same thing from a UI building point of view.

- [x] Uses native HTML functionality
- [x] Keyboard navigable
- [x] Nice animations

### CODE REVIEW

File: components/control/Input/Input.css [hunk 1]
```diff
@@ -20,6 +19,9 @@ diamond-input {
 
   &:focus-within {
     --_background: var(--diamond-input-background-focus);
+    /* https://css-tricks.com/copy-the-browsers-native-focus-styles/ */
```
Comment (DanWebb) [inline components/control/Input/Input.css:22]: Should this be on :has(input:focus-visible)? I wouldn't expect the ring to show when I click on an input, just the border to change

File: components/control/RadioCheckbox/RadioCheckbox.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  render() {
+    return html`
+      <span part="indicator">
+        <slot name="icon">
+          <svg
+            fill="none"
+            viewBox="0 0 24 24"
+            height="24"
+            width="24"
+            xmlns="http://www.w3.org/2000/svg"
```
Comment (DanWebb) [inline components/control/RadioCheckbox/RadioCheckbox.ts:23]:
```suggestion
```

xmlns is only needed if the SVG isn't inline

File: components/control/RadioCheckbox/RadioCheckbox.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            part="fallback-icon"
+          >
+            <path
+              xmlns="http://www.w3.org/2000/svg"
```
Comment (DanWebb) [inline components/control/RadioCheckbox/RadioCheckbox.ts:27]:
```suggestion
```

**Review by DanWebb — CHANGES_REQUESTED** [review]

File: components/control/Input/Input.css (hunk 1 above)
Comment (gavmck) [reply components/control/Input/Input.css:22]: You always get a ring on inputs, that's the standard behaviour unless we override it.
I still think your idea is better though, will prevent conflicts with buttons in suffix

**Review by DanWebb — APPROVED** [review]

**Pull request merged**